tests/backup/backup_rancher_test.go

```diff
@@ -265,3 +271,534 @@ var _ = Describe("{SingleNamespaceBackupRestoreToNamespaceInSameAndDifferentProj
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			backupLocationProviders := getProviders()
+			for _, provider := range backupLocationProviders {
+				credName = fmt.Sprintf("%s-cred-%v", provider, RandomString(10))
+				credUid = uuid.New()
+				err := CreateCloudCredential(provider, credName, credUid, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] as provider %s", credName, orgID, provider))
+				customBackupLocationName = fmt.Sprintf("%s-backup-location-%v", provider, RandomString(10))
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = customBackupLocationName
+				err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, credUid, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+			}
+		})
+
+		Step("Registering application clusters for backup", func() {
+			log.InfoD("Registering application clusters for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateApplicationClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Creating 2 rancher projects in source cluster", func() {
+			log.InfoD("Creating 2 rancher projects in source cluster")
+			for i := 0; i < 2; i++ {
+				sourceProject = fmt.Sprintf("source-project-%v-%v", RandomString(10), i)
+				_, err = Inst().S.(*rke.Rancher).CreateRancherProject(sourceProject, "new project", rancherActiveCluster)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating rancher project %s in source cluster", sourceProject))
+				sourceProjectID, err = Inst().S.(*rke.Rancher).GetProjectID(sourceProject)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Getting Project ID for project %s", sourceProject))
+				sourceProjectList = append(sourceProjectList, sourceProject)
+				sourceProjectIDList = append(sourceProjectIDList, sourceProjectID)
+			}
+		})
+
+		Step("Adding namespaces to project in source cluster and taking backup", func() {
+			log.InfoD("Adding namespaces to project in source cluster and taking backup")
+			log.Infof("Adding half of the namespaces to first project in source cluster")
+			err = Inst().S.(*rke.Rancher).AddNamespacesToProject(sourceProjectList[0], appNamespaces[0:len(appNamespaces)/2])
```

Comment:
Can we define a new variable for appNamespaces[0:len(appNamespaces)/2] and use it everywhere else?

Reply:
done